### PR TITLE
Edit release-process.md for clarity

### PR DESCRIPTION
**Pull request by arcalinea**
For #2160

### CODE REVIEW

File: doc/release-process.md [hunk 1]
```diff
@@ -53,7 +53,9 @@ In `configure.ac` and `clientversion.h`:
 If this release changes the behavior of the protocol or fixes a serious bug, we may
 also wish to change the `PROTOCOL_VERSION` in `version.h`.
 
-Build and commit to update versions, and then perform the following command:
+Build, then commit these changes to update versions.
```
Comment (str4d) [inline doc/release-process.md]: Needs to be "commit then build" so that the built binary doesn't include `-dirty`.

File: doc/release-process.md [hunk 2]
```diff
@@ -101,8 +103,9 @@ Run `./fetch-params.sh`.
 
 ## D. Make tag for the newly merged result
 
-In this example, we ensure master is up to date with the
-previous merged PR, then:
+Checkout master and pull the latest version to ensure master is up to date with the release PR which has recently been reviewed and merged. Check the last commit on the local and remote versions of master to make sure they are the same.
```
Comment (str4d) [inline doc/release-process.md]: - `s/which has recently been reviewed and merged/merged in B4/`
- Break the line?

**Comment by arcalinea** [comment]
@str4d incorporated your comments

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**